devito/types/grid.py

```diff
@@ -381,7 +394,15 @@ def dimensions(self):
 
     @property
     def dimension_map(self):
-        return {d.parent: d for d in self.dimensions}
+        dim_map = {}
```

Comment:
just switch ``d.parent`` to ``d.root`` and the one linear should work again

Reply:
Great, ta.